extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.hibernate.reactive.runtime;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.hibernate.reactive.stage.Stage;
+
+import io.quarkus.arc.DefaultBean;
+import io.smallrye.mutiny.Uni;
+
+@ApplicationScoped
+public class ReactiveSessionProducer {
+
+    @Inject
+    private Stage.SessionFactory reactiveSessionFactory;
+
+    @Inject
+    private Mutiny.SessionFactory mutinySessionFactory;
+
+    @Produces
+    @RequestScoped
+    @DefaultBean
+    public CompletionStage<Stage.Session> stageSession() {
+        return reactiveSessionFactory.openSession();
+    }
+
+    @Produces
+    @RequestScoped
+    @DefaultBean
+    public Uni<Mutiny.Session> mutinySession() {
+        return mutinySessionFactory.openSession();
+    }
+
+    public void disposeStageSession(@Disposes CompletionStage<Stage.Session> reactiveSession) {
+        reactiveSession.whenComplete((s, t) -> {
+            if (s != null)
+                s.close();
+        });
+    }
+
+    public void disposeMutinySession(@Disposes Uni<Mutiny.Session> reactiveSession) {
+    	// TODO: @AGG this is not working, need to check w/ Clement to figure out the proper way
```

Comment:
I doubt the version for `CompletionStage` works either: this creates a new `Uni`/`CompletionStage` that must be used in order for it to be terminated. This has no side-effect.